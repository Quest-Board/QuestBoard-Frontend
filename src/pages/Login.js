import React, { Component } from "react";
import {Redirect} from "react-router-dom";

export default class Login extends Component {
    password;
    address;

    constructor(props){
        super(props);
        this.state={redirect:null};
    }

    //send login attempt with no body, will return ok if the 
    //user has a valid cookie from the last login
    CheckForCookie=()=> {
        fetch('https://coms-319-t15.cs.iastate.edu/api/board/create?boardname=testBoard', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
          })
          .then((response) => {
            if (response.ok){
                this.setState({redirect: "/User/Board"})
            }
          })
    }

    componentDidMount() {
        this.CheckForCookie();
    }

    LoginCall=(event)=>{
        event.preventDefault();
        return fetch('https://coms-319-t15.cs.iastate.edu/api/account/login', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
              email: this.address,
              password: this.password
            })
          })
          .then((response) => {
            if (response.ok){
                console.log("ok");
                this.setState({redirect: "/User/Board"})
            }
            else window.alert("Username or password is incorrect.")
          })
    }

    UserChangeHandler=(event)=>{
        this.address=event.target.value;
    }

    PassChangeHandler=(event)=>{
        this.password=event.target.value;
    }

    render() {
        if(this.state.redirect){
            return <Redirect to={this.state.redirect}/>
        }
        return (
            <form onSubmit={this.LoginCall}>
                <h3>Sign In</h3>

                <div className="form-group">
                    <label>Email address</label>
                    <input type="email" onChange={this.UserChangeHandler} className="form-control" placeholder="Enter email" />
                </div>

                <div className="form-group">
                    <label>Password</label>
                    <input type="password" onChange={this.PassChangeHandler} className="form-control" placeholder="Enter password" />
                </div>

                <div className="form-group">
                    <div className="custom-control custom-checkbox">
                        <input type="checkbox" className="custom-control-input" id="customCheck1" />
                        <label className="custom-control-label" htmlFor="customCheck1">Remember me</label>
                    </div>
                </div>

                <button type="submit" className="btn btn-primary btn-block" >Submit</button>
                <p className="forgot-password text-right">
                    Forgot <a href="#">password?</a>
                </p>
            </form>
        );
    }
}