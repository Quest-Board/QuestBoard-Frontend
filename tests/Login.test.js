import 'regenerator-runtime/runtime';
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../src/pages/Login';
import { BrowserRouter } from 'react-router-dom';


global.fetch = jest.fn(() => Promise.resolve());

test('Login snapshot test', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Login/>
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});