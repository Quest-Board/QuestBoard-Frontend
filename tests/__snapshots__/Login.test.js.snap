// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Sign In
  </h3>
  <div
    className="form-group"
  >
    <label>
      Email address
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter email"
      type="email"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      Password
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Enter password"
      type="password"
    />
  </div>
  <div
    className="form-group"
  >
    <div
      className="custom-control custom-checkbox"
    >
      <input
        className="custom-control-input"
        id="customCheck1"
        type="checkbox"
      />
      <label
        className="custom-control-label"
        htmlFor="customCheck1"
      >
        Remember me
      </label>
    </div>
  </div>
  <button
    className="btn btn-primary btn-block"
    id="LoginButton"
    type="submit"
  >
    Submit
  </button>
  <p
    className="forgot-password text-right"
  >
    Forgot 
    <a
      href="#"
    >
      password?
    </a>
  </p>
</form>
`;
